arch/x86/core/x86_mmu.c

```diff
@@ -136,3 +136,155 @@ void _x86_mmu_set_flags(void *ptr, size_t size, u32_t flags, u32_t mask)
+		total_partitions = mem_domain->num_partitions;
+		__ASSERT(total_partitions <= CONFIG_MAX_DOMAIN_PARTITIONS,
+			 "invalid partitions");
+
+		/* Interate over all the partitions for the given mem_domain
+		 * For x86: interate over all the partitions and set the
+		 * required flags in the correct MMU page tables.
+		 */
+		for(partition_index = 0;
+		    partition_index <= total_partitions;
+		    partition_index++){
+
+			/* Get the partition info */
+			partition = mem_domain->partitions[partition_index];
+			/* Set the partition attributes */
+			_x86_mmu_set_flags((void *)partition.start,
+					   partition.size,
+					   partition.attr,
+					   K_MEM_PARTITION_PERM_MASK);
+
+		}
+
+	} while(0);
+	return;
+}
+
+/* Load only one parttion of the thread */
+void x86_load_mmu_mem_partition(struct k_thread *thread, u32_t partition_index)
+{
+	u32_t total_partitions;
+	struct k_mem_partition partition;
+	struct k_mem_domain *mem_domain;
+	mem_domain = thread->mem_domain_info.mem_domain;
+
+	do {
+		/* If mem_domain doesn't point to a valid location return.*/
+		if (mem_domain == NULL) {
+			break;
+		}
+
+		/* Get the total number of partitions.
+		 * Block if the partitions exceed the max.*/
+		total_partitions = mem_domain->num_partitions;
+		__ASSERT(partition_index <= total_partitions,
+			 "invalid partitions");
+
+		/* Interate over all the partitions for the given mem_domain
+		 * For x86: interate over all the partitions and set the
+		 * required flags in the correct MMU page tables.
+		 */
+
+		/* Get the partition info */
+		partition = mem_domain->partitions[partition_index];
+		/* Set the partition attributes */
+		_x86_mmu_set_flags( (void*)partition.start,
+				   partition.size,
+				   partition.attr,
+				   K_MEM_PARTITION_PERM_MASK);
+
+
+	} while(0);
+	return;
+}
+
+
+/* Destroy or reset the mmu page tables when necessary.
+ * Needed when either swap takes place or k_mem_domain_destroy is called.  */
+void x86_destroy_all_mem_partitions(struct k_thread *thread)
+{
+	u32_t partition_index;
+	u32_t total_partitions;
+	struct k_mem_partition partition;
+
+	/* Get the total number of partitions */
+	do {
+		if (thread->mem_domain_info.mem_domain == NULL) {
+			break;
+		}
+		total_partitions =
+			thread->mem_domain_info.mem_domain->num_partitions;
+		__ASSERT(total_partitions <= CONFIG_MAX_DOMAIN_PARTITIONS,
+			 "invalid partitions");
+
+		for(partition_index = 0;
+		    partition_index <= total_partitions;
+		    partition_index++){
+
+			/* can ommit the next line if the macros are
+			   properly defined. */
+			partition = thread->mem_domain_info.mem_domain-> \
+				partitions[partition_index];
+			_x86_mmu_set_flags( (void*)partition.start,
+					   partition.size,
+					   K_MEM_PARTITION_P_RW_U_NA,
+					   K_MEM_PARTITION_PERM_MASK);
+		}
+
+	}while(0);
+	return;
+}
+
+/* Reset/destroy one partition spcified in the argument of the API. */
+void x86_destroy_mem_partition(struct k_thread *thread, u32_t partition_index)
```

Comment:
we don't need this function, the only time we need to update the page tables is on swap and when dropping to user mode, and in those cases we just update the entire domain